Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -639,35 +639,43 @@ extension BrowserViewController: WKNavigationDelegate {
     }
     
     func webView(webView: WKWebView, didCommitNavigation navigation: WKNavigation!) {
-        urlBar.updateURL(webView.URL);
-        toolbar.updateBackStatus(webView.canGoBack)
-        toolbar.updateFowardStatus(webView.canGoForward)
-        showToolbars(animated: false)
+        if let tab = tabManager.selectedTab {
+            if tab.webView == webView {
```

Comment:
If you want, you could remove a level of nesting on these checks by combining these:
```
if tabManager.selectedTab?.webView == webView
```